Classes/Security/FilePermissionsAspect.php

```diff
@@ -57,12 +71,11 @@ protected function respectStorage(Matcher $matcher) {
 	}
 
 	/**
-	 * @param Matcher $matcher
+	 * @param Query $query
+	 * @param ConstraintInterface|NULL $constraints
 	 * @return void
 	 */
-	protected function respectFileMounts(Matcher $matcher) {
-		$matcher->setLogicalSeparatorForLike(Matcher::LOGICAL_OR);
-
+	protected function respectFileMounts(Query &$query, &$constraints) {
```

Comment:
i use PHP 5.3.n on my server. i had that problem some time ago within a feature request for TYPO3 CMS core. after some explanations of a core team member i got it to work without the &. unfortunatelly not with this pull request, but i am sure there is something i did wrong on my side. i will test it!

it would be more than fine with me if you would merge this pull request (and the EXT:vidi pull request it depends on) without the &.